File: backend/services/imagekit_service.py
import logging
from imagekitio import ImageKit
from config import IMAGEKIT_PUBLIC_KEY, IMAGEKIT_PRIVATE_KEY, IMAGEKIT_URL_ENDPOINT

logger = logging.getLogger(__name__)

ik = ImageKit(
    public_key=IMAGEKIT_PUBLIC_KEY,
    private_key=IMAGEKIT_PRIVATE_KEY,
    url_endpoint=IMAGEKIT_URL_ENDPOINT
)

async def upload_file(file_bytes: bytes, filename: str) -> str:
    """
    Upload file bytes to ImageKit and return the URL.
    """
    try:
        upload = ik.upload(
            file=file_bytes,
            file_name=filename,
        )
        return upload.url
    except Exception as e:
        logger.error(f"Error uploading to ImageKit: {e}")
        raise e

File: backend/src/services/imagekit_service.ts
import ImageKit from 'imagekit';
import dotenv from 'dotenv';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

dotenv.config({ path: path.join(__dirname, '../../../.env') });

const ik = new ImageKit({
  publicKey: process.env.IMAGEKIT_PUBLIC_KEY || '',
  privateKey: process.env.IMAGEKIT_PRIVATE_KEY || '',
  urlEndpoint: process.env.IMAGEKIT_URL_ENDPOINT || '',
});

export const uploadFile = async (fileBuffer: Buffer | string, fileName: string, folder: string = 'headshots'): Promise<string> => {
  try {
    const upload = await ik.upload({
      file: fileBuffer,
      fileName: fileName,
      folder: folder,
    });
    return upload.url;
  } catch (error) {
    console.error('Error uploading to ImageKit:', error);
    throw error;
  }
};

export interface ImageVariants {
  youtube: string;
  shorts: string;
  square: string;
}

export const getVariants = (baseUrl: string | null | undefined): ImageVariants | null => {
  if (!baseUrl) return null;
  return {
    youtube: `${baseUrl}?tr=w-1280,h-720,c-maintain_ratio,fo-auto`,
    shorts: `${baseUrl}?tr=w-1080,h-1920,c-maintain_ratio,fo-auto`,
    square: `${baseUrl}?tr=w-1080,h-1080,c-maintain_ratio,fo-auto`,
  };
};
